### PR TITLE
feat(evaluation) 2/15: test-set parsing and file_id normalisation

### PR DESCRIPTION
**Pull request by Ahmath-Gadji**
Part **2 of 15** of the split of #811. Targets `eval/01-config-and-models` (#812) — merge that first.

## What

The first two pieces of `core/evaluation/`, both pure functions with no I/O.

- **`parse_testset`** — reads the admin-uploaded CSV (`question,expected_answer,expected_file_ids`) into `EvalTestCase` rows, under a row cap. `expected_file_ids` is optional and `;`-separated.
- **`sanitize_file_id`** — maps a corpus filename onto an id the indexing API accepts.

## Notable

- **Why `identity.py` is its own module.** The indexing API allows only `[A-Za-z0-9._:-]` in a `file_id`, so a corpus file cannot be uploaded under a raw human filename. *Both* sides of the ground-truth comparison must therefore normalise: the uploader (part 9) and the metrics (part 4). A test set naming `A B.pdf` matches the stored `A_B.pdf` only because one function owns that mapping.
- **Validation happens at upload, not at run time.** A dataset whose CSV is malformed should fail while the admin is still looking at the upload dialog, not thirty minutes into an indexing phase. Part 7 calls this from `create_dataset` for that reason.

## Testing

12 new unit tests over the parser: header validation, the optional third column, blank and malformed rows, the row cap, and encoding. `ruff`, format check and the layer-import guard pass.

## Summary by CodeRabbit

* **New Features**
  * Added evaluation utilities to parse and strictly validate uploaded prompt-style CSV test sets.
  * Supports UTF-8 input (including BOM), header normalization, optional expected file IDs, and ignores fully blank rows.
  * Introduced consistent file ID sanitization for indexing compatibility.
* **Bug Fixes**
  * Improved validation behavior with clear, specific error codes for empty/invalid inputs, missing or duplicate columns, invalid encodings, and exceeding the row limit.
* **Tests**
  * Added unit tests covering successful parsing, edge cases, and error scenarios.

### CODE REVIEW

File: openrag/core/evaluation/testset.py [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+    except UnicodeDecodeError as exc:
+        raise ValidationError(
+            "Test set must be UTF-8 encoded CSV.",
+            code="EVAL_TESTSET_ENCODING",
+            status_code=400,
+        ) from exc
+
+
+def _split_file_ids(cell: str) -> tuple[str, ...]:
+    return tuple(part.strip() for part in cell.split(FILE_ID_SEPARATOR) if part.strip())
```
[the remark this answers is not in the document]
Comment (Ahmath-Gadji) [reply openrag/core/evaluation/testset.py:45]: Not applying this one — the match already works, and the normalisation you're pointing at happens a layer down.

The reason it isn't visible here is an artifact of how this feature was split for review: `metrics.py` lands in #815, three PRs downstream, so it isn't in this diff. `summarize()` sanitises **both** sides at comparison time:

```python
# core/evaluation/metrics.py
if case.has_ground_truth_sources:
    expected = {sanitize_file_id(name) for name in case.expected_file_ids}
    matched = [rank for rank, (_, ids) in enumerate(documents, start=1) if ids & expected]
```

and the retrieved side goes through the same function in `_retrieved_documents()`. So a test set naming `A B.pdf` does match the stored `A_B.pdf` today — `test_ground_truth_matches_the_original_filename_when_the_file_id_was_sanitised` and `test_matching_survives_file_id_sanitisation_on_either_side` in #815 are exactly that case, from both directions.

Given that, moving it into the parser would be redundant for matching and lossy for everything else. `EvalTestCase.expected_file_ids` is not only a matching key: it is copied verbatim onto `EvalCaseResult.expected_file_ids`, which the API returns in the run detail response. Sanitising at parse time would mean an author who wrote `A B.pdf` gets `A_B.pdf` echoed back, with no record of what they actually wrote — which is the wrong direction precisely when someone is debugging a row that *didn't* match. `tests/evaluation/README.md` tells authors to name files as they appear on disk, and this is what makes that true.

One place the parser is the right home would be if matching moved to a normalised key at ingest, but then it should replace the sanitisation in `summarize`, not duplicate it. Keeping a single normalisation point, at the comparison, is the cheaper invariant to hold.

**Review by hedhoud — CHANGES_REQUESTED** [review]
The parser is going in the right direction, but I found two cases that can make the evaluation results unreliable.

Different filenames can currently produce the same file ID. For example, `A B.pdf` and `A_B.pdf` both become `A_B.pdf`. If both are in the corpus, one upload will fail and the metrics may match the wrong file. The generated ID should stay unique, or the upload should reject that collision with a clear error.

Malformed CSV rows can also be silently dropped. A row with only `expected_file_ids`, or a row with more values than its header, can be accepted without telling the user that some data was ignored. Please reject any non-empty malformed row and add tests for these cases. It would also be useful to report the CSV reader’s real line number when blank lines are present.

Once these cases are covered, this should be safe to move forward.
